src/infrastructure/Http.ts

```diff
@@ -14,36 +14,40 @@
  * limitations under the License.
  */
 
-import {Observable, of as observableOf} from 'rxjs';
-import {map} from 'rxjs/operators';
+// tslint:disable-next-line: ordered-imports
+import {from as observableFrom, Observable, of as observableOf, throwError} from 'rxjs';
+import {catchError, map} from 'rxjs/operators';
 import {NetworkType} from '../model/blockchain/NetworkType';
-import {NetworkHttp} from './NetworkHttp';
+import { NodeRoutesApi } from './api/apis';
 import { QueryParams } from './QueryParams';
 /**
  * Http extended by all http services
  */
 export abstract class Http {
-    private networkHttp: NetworkHttp;
-    private networkType: NetworkType;
+    protected readonly url: string;
+    protected networkType: NetworkType;
 
     /**
      * Constructor
-     * @param url
-     * @param networkHttp
+     * @param url Base catapult-rest url
+     * @param networkType
      */
-    constructor(networkHttp?: NetworkHttp) {
-        if (networkHttp) {
-            this.networkHttp = networkHttp;
+    constructor(url: string, networkType?: NetworkType) {
+        if (networkType) {
+            this.networkType = networkType;
         }
+        this.url = url;
     }
 
     getNetworkTypeObservable(): Observable<NetworkType> {
         let networkTypeResolve;
-        if (this.networkType == null) {
-            networkTypeResolve = this.networkHttp.getNetworkType().pipe(map((networkType) => {
-                this.networkType = networkType;
-                return networkType;
-            }));
+        if (!this.networkType) {
+            networkTypeResolve = observableFrom(new NodeRoutesApi(this.url).getNodeInfo()).pipe(
+                map(({body}) => {
+                    return body.networkIdentifier;
+                }),
+                catchError((error) =>  throwError(this.errorHandling(error))),
+            );
```

Comment:
Every time getNetworkTypeObservable is called a new NodeRoutesApi is created and the rest endpoint is called. This is not very efficient and it may ban the server. I would move the networkTypeResolve to a read-only field and create it up in the constructor.  If cached, the user has less need for hard-coding the network type in the constructor and leaves the HTTPs to handle the network type efficiently. 

Java has an [observable.cache](url) method to cache the result in memory. 

https://github.com/nemtech/nem2-sdk-java/blob/master/sdk-okhttp-client/src/main/java/io/nem/sdk/infrastructure/okhttp/RepositoryFactoryOkHttpImpl.java#L66

Once we have the factory, the cache will work much better as the cached value will be reused between multiple Http Objects (Transaction, Account, etc...).

Reply:
Can put the refactoring into a separate sdk refactor task. Also the cash would probably be better to put into the implementation e.g. client wallet etc to be able to provide more control.